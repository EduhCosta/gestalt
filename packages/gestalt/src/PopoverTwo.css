.customClass {
  background: rgb(0 0 0 / 0.061);
}

.arrow,
.arrow::before {
  background: inherit;
  height: 8px;
  position: absolute;
  width: 8px;
}

.arrow {
  visibility: hidden;
}

.arrow::before {
  content: '';
  transform: rotate(45deg);
  visibility: visible;
}

.tooltip[data-popper-placement^='top']>.arrow {
  bottom: -4px;
}

.tooltip[data-popper-placement^='bottom']>.arrow {
  top: -4px;
}

.tooltip[data-popper-placement^='left']>.arrow {
  right: -4px;
}

.tooltip[data-popper-placement^='right']>.arrow {
  left: -4px;
}
