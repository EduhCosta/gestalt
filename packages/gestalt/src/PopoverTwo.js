// @flow strict

import { type Node, useState, Fragment } from 'react';
import { usePopper } from 'react-popper';
import Box from './Box.js';

type Props = {|
  /**
   * Prop description.
   */
  accessibilityLabel?: string,
|};

function Example() {
  const [referenceElement, setReferenceElement] = useState(null);
  const [popperElement, setPopperElement] = useState(null);
  const [arrowElement, setArrowElement] = useState(null);
  const { styles, attributes } = usePopper(referenceElement, popperElement, {
    modifiers: [{ name: 'arrow', options: { element: arrowElement } }],
  });

  console.log(attributes);

  return (
    <Fragment>
      <button type="button" ref={setReferenceElement}>
        Reference element
      </button>

      <Box
        ref={setPopperElement}
        style={styles.popper}
        dangerouslySetInlineStyle={{ __style: { backgroundColor: 'red' } }}
      >
        Popper element
        <Box ref={setArrowElement} style={styles.arrow} />
      </Box>
    </Fragment>
  );
}

/**
 * [PopoverTwo] https://gestalt.pinterest.systems/web/popovertwo component should be used for ... on the page.
 * ![PopoverTwo light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/PopoverTwo.spec.mjs-snapshots/PopoverTwo-chromium-darwin.png)
 * ![PopoverTwo dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/PopoverTwo-dark.spec.mjs-snapshots/PopoverTwo-dark-chromium-darwin.png)
 */
export default function PopoverTwo({ accessibilityLabel }: Props): Node {
  return <Example />;
}
