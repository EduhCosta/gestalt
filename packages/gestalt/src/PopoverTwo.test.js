// @flow strict
import { create } from 'react-test-renderer';
import PopoverTwo from './PopoverTwo.js';

describe('PopoverTwo', () => {
  it('renders', () => {
    const tree = create(<PopoverTwo name="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an accessibility label', () => {
    const tree = create(<PopoverTwo accessibilityLabel="Test Accessibility Label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
