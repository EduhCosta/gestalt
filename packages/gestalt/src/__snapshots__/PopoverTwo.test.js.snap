// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverTwo renders 1`] = `
<div
  className="box"
>
  <div
    className="customClass"
  />
</div>
`;

exports[`PopoverTwo renders an accessibility label 1`] = `
<div
  aria-label="Test Accessibility Label"
  className="box"
>
  <div
    className="customClass"
  />
</div>
`;
